### ISSUE TITLE
UTF8_ToNative() turns null C# strings into empty C strings, instead of into null pointers.

### ISSUE THREAD

**Issue by holmak** [flibitijibibo/SDL2-CS#125]
**Expected behavior:** When you pass a null C# string to an SDL2-CS wrapper function, it is converted into a null pointer and sent to the actual SDL function.
**Actual behavior:** The wrapper functions convert null C# strings into _empty_ C strings.

The function `SDL_OpenAudioDevice()` takes a device name as its first argument. The device name can be either the name of an audio device, or `NULL` to let SDL choose an audio device. When you call the C# binding, `SDL.SDL_OpenAudioDevice(null, ...)`, that binding function uses `UTF8_ToNative()` to convert the C# string UTF8 C string:

```
public static uint SDL_OpenAudioDevice(
	string device,
	int iscapture,
	ref SDL_AudioSpec desired,
	out SDL_AudioSpec obtained,
	int allowed_changes
) {
	return INTERNAL_SDL_OpenAudioDevice(
		UTF8_ToNative(device),
		iscapture,
		ref desired,
		out obtained,
		allowed_changes
	);
}
```

The problem is that UTF8_ToNative is implemented as:

```
internal static byte[] UTF8_ToNative(string s)
{
	// Add a null terminator. That's kind of it... :/
	return System.Text.Encoding.UTF8.GetBytes(s + '\0');
}
```

When `s` is null, `s + '\0\'` produces the C# string `"\0"`, which `GetBytes()` converts to an empty C string. This causes a problem in SDL_OpenAudioDevice, because `""` (the empty string) is an invalid audio device name -- and I meant to pass NULL.

I believe UTF8_ToNative() should be implemented as:

```
internal static byte[] UTF8_ToNative(string s)
{
    // Add a null terminator. That's kind of it... :/
    if (s != null)
    {
        return System.Text.Encoding.UTF8.GetBytes(s + '\0');
    }
    else
    {
        return null;
    }
}
```

This would produce the correct behavior for SDL_OpenAudioDevice(), but I'm not sure it is what all usages of `UTF8_ToNative()` expect the UTF8 conversion to do.

**Comment by flibitijibibo** [comment]
That looks right to me, go ahead and send it as a PR and I’ll merge it in.

**Comment by flibitijibibo** [comment]
Solved by #126
